src/classes/CallableDispatchService_ERR.cls

```diff
@@ -0,0 +1,196 @@
+    public static final String PARAM_SOBJECTS = 'SObjects';
+
+    public static final String PARAM_CONTEXT_NAME = 'Context';
+    public static final String PARAM_ERR_MSGS_BY_RECORDID = 'ErrorMsgsByRecordId';
+    public static final String PARAM_DML_RESULTS = 'DmlResults';
+    public static final String PARAM_MESSAGE = 'Message';
+    public static final String PARAM_EXCEPTION = 'Exception';
+
+    /**
+     * @description Dispatches invocations to the Error (ERR) Handling functionality from the Callable API to different methods
+     * based on an "ActionType" parameter
+     * @param CallableApiParameters paramService Parameteters passed to the Callable API as part of it's invocation
+     * @return Object The result of the dispatch action's invocation
+     */
+    public static Object dispatch(CallableApiParameters paramService) {
+
+        String dispatchAction = paramService.getString(PARAM_DISPATCH_ACTION_TYPE);
+
+        String cleanDispatchAction = (String.isBlank(dispatchAction) ? '' : dispatchAction.toLowerCase());
+
+        // Pulling Context out of switch since all the dispatch actions need the Context parameter
+        String context = paramService.getString(PARAM_CONTEXT_NAME);
+
+        switch on (cleanDispatchAction) {
+            when 'adderror' { // Add Error
+                addError(   context,
+                            paramService.getString(PARAM_MESSAGE)
+                );
+
+            } when 'addexception' { // Add Exception
+                addException(   context,
+                                paramService.getException(PARAM_EXCEPTION)
+                );
+
+            } when 'addrecorderrors' { // Add Record Errors
+                addRecordErrors(    context,
+                                    paramService.getSObjects(PARAM_SOBJECTS),
+                                    paramService.getStringsById(PARAM_ERR_MSGS_BY_RECORDID)
+                );
+
+            } when 'addrecorddmlerrors' { // Add Record Dml Errors
+                addRecordDmlErrors( context,
+                                    paramService.getSObjects(PARAM_SOBJECTS),
+                                    paramService.getObjects(PARAM_DML_RESULTS)
+                );
+
+            } when 'clearrecorderrors' { // Clear Record Errors
+                clearRecordErrors(  context,
+                                    paramService.getSObjects(PARAM_SOBJECTS),
+                                    paramService.getString(PARAM_MESSAGE)
+                );
+
+            } when else { // Invalid Action Type / Action Type Not implemented
+                throw new MalformedErrorDispatchMethodApiException(
+                    String.format(System.Label.CallableApiMethodNotImplemented, new List<String> { 'err.dispatchaction => ' + dispatchAction})
+                );
+            }
+        }
+
+        // if the action does not return a value then return success
+        return true;
```

Comment:
Perfect. That's how the CallableApi does it as well. This way your individual methods can return when they need to, but this acts as a catch all if there is no explicit return.